source/fundamentals/auth.txt

```diff
@@ -189,59 +189,104 @@ username and password to authenticate your user.
 
 .. important::
 
-   The MONGODB-AWS authentication mechanism is only available in MongoDB
+   The MONGODB-AWS authentication mechanism is available only in MongoDB
    versions 4.4 and later.
 
 The ``MONGODB-AWS`` authentication mechanism uses your Amazon Web Services
 Identity and Access Management (AWS IAM) credentials to authenticate your
 user.
 
-To specify the ``MONGODB-AWS`` authentication mechanism, perform the
-following:
+To connect to a MongoDB instance with ``MONGODB-AWS`` authentication enbaled,
+specify the ``MONGODB-AWS`` authentication mechanism.
 
-- Assign the ``AuthMechanism`` option the value ``MONGODB-AWS``
-- Assign the ``Username`` option the value of your ``accessKeyID``
-- Assign the ``Password`` option the value of your ``secretAccessKey``
+The driver checks for you credentials in the following sources in order:
 
-.. code-block:: go
-   :emphasize-lines: 3, 5-6
+1. Connection string
+#. Environment variables
+#. Web identity token file
+#. AWS ECS endpoint specified in ``AWS_CONTAINER_CREDENTIALS_RELATIVE_URI``
+#. AWS EC2 endpoint. For more information, see `IAM Roles for Tasks
+   <https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html>`_.
 
-   var accessKeyID, secretAccessKey string
-   awsCredential := options.Credential{
-      AuthMechanism: "MONGODB-AWS",
-      AuthSource: "<authenticationDb>",
-      Username: "<accessKeyID>",
-      Password: "<secretAccessKey>",
-   }
-   awsIAMClient, err := mongo.Connect(
-      context.TODO(),
-      options.Client().SetAuth(awsCredential))
-   if err != nil {
-      panic(err)
-   }
-   _ = awsIAMClient
+.. important::
+   
+   The driver only reads the credentials from the first method that it detects
+   in the order as given by the preceding list. For example, if you specify 
```

Comment:
C: a little confusing

```suggestion
   The driver obtains credentials from the first source that they are detected. For example, if you specify 
```

Reply:
Adjusted the wording a bit, let me know if that is better